Whisky/Views/Bottle/Pins/PinCreationView.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  PinCreationView.swift
+//  Whisky
+//
+//  This file is part of Whisky.
+//
+//  Whisky is free software: you can redistribute it and/or modify it under the terms
+//  of the GNU General Public License as published by the Free Software Foundation,
+//  either version 3 of the License, or (at your option) any later version.
+//
+//  Whisky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+//  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+//  See the GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License along with Whisky.
+//  If not, see https://www.gnu.org/licenses/.
+//
+
+import SwiftUI
+import UniformTypeIdentifiers
+import WhiskyKit
+
+struct PinCreationView: View {
+    let bottle: Bottle
+    @State private var newPinName: String = ""
+    @State private var newPinURL: URL = BottleData.defaultBottleDir
+    @State private var isValid: Bool = false
+    @State private var isDuplicate: Bool = false
+    @Environment(\.dismiss) private var dismiss
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            Text("pin.title")
+                .bold()
+            Divider()
+            HStack {
+                Text("pin.name")
+                    .frame(maxWidth: /*@START_MENU_TOKEN@*/.infinity/*@END_MENU_TOKEN@*/, alignment: .leading)
```

Comment:
you have some helper tags here. You will see `.infinity` highlighted in Xcode. you should double click on it to select it and get rid of these helper tags

Reply:
Arg, rookie mistake. Fixed in latest commit.